src/DFG.cpp

```diff
@@ -175,6 +175,44 @@ void DFG::combine(string t_opt0, string t_opt1) {
   }
 }
 
+void DFG::combineIter(list<string>* cPattern, list<DFGNode*>* cPatternNodes) {
+  string t_opt = cPattern->front();
+  cPattern->pop_front();
```

Comment:
`cPattern` is always popped. How could we pattern match one string pattern with multiple for loops (if there are)?